### PR TITLE
SWS-261 Remove abusing usage of channels

### PR DESCRIPTION
**Pull request by jotak**
Simple synchronization between goroutines is better done through async.WaitGroup
Also removed some async calls where it seems like overkill

### CODE REVIEW

**Comment by jotak** [comment]
@burmanm WDYT? Better like that?

File: prometheus/client.go [hunk 1]
```diff
@@ -86,17 +86,12 @@ func (in *Client) GetSourceServices(namespace string, servicename string) (map[s
 // GetServiceMetrics returns the Health and Metrics related to the provided service identified by its namespace and service name.
 // Health might be nil when unavailable
 func (in *Client) GetServiceMetrics(namespace string, servicename string, duration time.Duration, step time.Duration,
-	rateInterval string) Metrics {
+	rateInterval string) (Metrics, Health) {
 
-	healthChan, metricsChan := make(chan *Health), make(chan Metrics)
-	go getServiceHealthAsync(in.api, namespace, servicename, healthChan)
-	go getServiceMetricsAsync(in.api, namespace, servicename, duration, step, rateInterval, metricsChan)
+	metrics := getServiceMetrics(in.api, namespace, servicename, duration, step, rateInterval)
```
Comment (burmanm) [inline prometheus/client.go]: These are no longer done in parallel. Intentional?

Comment (jotak) [reply prometheus/client.go]: Yes, I changed my approach from "run everything possible in parallel" to something more reasonable I think, like "run what's make most sense in parallel". Though I'm not yet 100% sure about what _makes most sense_.

**Comment by jotak** [comment]
Closing, I will reopen a PR with a separate JIRA for that, but this was mixing two things: channels work and separating Health. Need to clean that up.

**Pull request closed without merging**